pkg/kp/rcstore/consul_store.go

```diff
@@ -305,38 +299,37 @@ func (s *consulStore) mutateRc(id fields.ID, mutator func(fields.RC) (fields.RC,
 func (s *consulStore) Watch(rc *fields.RC, quit <-chan struct{}) (<-chan struct{}, <-chan error) {
 	updated := make(chan struct{})
 	errors := make(chan error)
-	var curIndex uint64 = 0
+	input := make(chan api.KVPairs)
+
+	targetPath := kp.RCPath(rc.ID.String())
+	go consulutil.WatchPrefix(targetPath, s.kv, input, quit, errors)
```

Comment:
Bleh. There's one other place in p2 that uses watch on get instead of list (checker.watchnodeservice). Should we also add a generic watchkey function for that?